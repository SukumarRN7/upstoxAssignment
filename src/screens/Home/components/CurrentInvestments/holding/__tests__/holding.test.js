import React from 'react';
import renderer from 'react-test-renderer';
import {Holding} from '../index';

describe('Holding', () => {
  const holdingData = {
    symbol: 'AAPL',
    avgPrice: 150,
    ltp: 160,
    quantity: 10,
  };

  test('renders correctly', () => {
    const component = renderer.create(<Holding holding={holdingData} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
