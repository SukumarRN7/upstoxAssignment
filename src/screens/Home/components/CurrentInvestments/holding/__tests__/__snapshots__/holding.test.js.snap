// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holding renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
    }
  }
>
  <View
    style={
      [
        {
          "borderBottomColor": "#ededed",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 20,
          "paddingVertical": 20,
        },
        undefined,
      ]
    }
  >
    <View>
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        AAPL
      </Text>
      <Text
        style={
          {
            "marginTop": 5,
          }
        }
      >
        10
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <Text>
        LTP : 
         
        <Text
          style={
            {
              "fontWeight": "bold",
            }
          }
        >
          ₹
           
          160
        </Text>
      </Text>
      <Text
        style={
          {
            "marginTop": 5,
          }
        }
      >
        PL : 
         
        <Text
          style={
            [
              {
                "fontWeight": "bold",
              },
              {
                "color": "green",
              },
            ]
          }
        >
          ₹
           
          100.00
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
